pkg/server/apiHandler.go

```diff
@@ -319,90 +196,35 @@ func (h *apiHandler) logs(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
 	}
-	msg := &logs.LogBundle{}
-	if err := proto.Unmarshal(b, msg); err != nil {
-		log.Printf("Failed to parse logbundle message: %v", err)
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+
+	status, err := logsProcess(h.manager, h.logChannel, *u, b)
+	if err != nil {
+		log.Printf("Failed to logsProcess: %v", err)
+		http.Error(w, http.StatusText(status), status)
 		return
 	}
-	eveVersion := msg.GetEveVersion()
-	image := msg.GetImage()
-	for _, entry := range msg.GetLog() {
-		entry := &common.FullLogEntry{
-			LogEntry:   entry,
-			Image:      image,
-			EveVersion: eveVersion,
-		}
-		var entryBytes []byte
-		if entryBytes, err = entry.Json(); err != nil {
-			log.Printf("Failed to marshal FullLogEntry message: %v", err)
-			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
-			return
-		}
-		select {
-		case h.logChannel <- entryBytes:
-		default:
-		}
-		err = h.manager.WriteLogs(*u, entryBytes)
-		if err != nil {
-			log.Printf("Failed to write log message: %v", err)
-			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-			return
-		}
-	}
-
-	// send back a 201
-	w.WriteHeader(http.StatusCreated)
+	w.WriteHeader(status)
 }
 
 func (h *apiHandler) newLogs(w http.ResponseWriter, r *http.Request) {
 	u := h.checkCertAndRecord(w, r)
 	if u == nil {
 		return
 	}
-	gr, err := gzip.NewReader(r.Body)
-	if err != nil {
-		log.Printf("error gzip.NewReader: %v", err)
-		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+	b, err := ioutil.ReadAll(r.Body)
```

Comment:
This strikes me as problematic. You are reading the entire body into memory here. What if the body is quite large? This can (and does) happen. With `gzip.NewReader` we are getting a stream, and so we can process each message at a time, without overloading it all into memory. We used to read it all, and then switched to streaming.

Reply:
You are right, thank you, fixed.

Reply:
Looks good.